org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/embedder/MavenImplTest.java

```diff
@@ -559,4 +560,22 @@ public MavenExecutionResult call(IMavenExecutionContext context, IProgressMonito
 
     assertFalse(result.getExceptions().toString(), result.hasExceptions());
   }
+
+  public void test486737_readProject() throws Exception {
```

Comment:
Igor requested tests both for MavenImpl #readProject and #readMavenProject, so here they are....